### PR TITLE
Contact fix

### PR DESCRIPTION
**Pull request by aaronk**
Updating the contact detail display, contact form and user / organization contact forms, mostly to allow the contact type to drive the fields that are displayed for the form.
Includes creation of UserContact in the api so that User and Organization contacts work the same way.

### CODE REVIEW

File: src/components/contact/IFXContactCreateEdit.vue [hunk 1]
```diff
@@ -1,31 +1,75 @@
...
     },
     updateContactOrganizations(organizations) {
       this.$set(this.item.organizations, ...organizations)
     },
+  },
+  computed: {
+    title() {
+      const createOrEdit = this.id ? 'Edit' : 'Create'
+      const type = this.localContactType ? `${this.localContactType} Contact` : ' Contact'
+      const id = this.item && this.item.name ? `for ${this.item.name}` : this.id || ''
```
Comment (luthian) [inline src/components/contact/IFXContactCreateEdit.vue:44]: We should start allowing [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) so we can do things like:
`this.item?.name`

File: src/components/contact/IFXContactCreateEdit.vue [hunk 2]
```diff
@@ -1,31 +1,75 @@
...
+    },
+    isEditing() {
+      return this.id !== null && this.id !== ''
+    }
+  },
+  mounted() {
+    if (this.id) {
+      this.$api.contact.getByID(this.id)
+        .then((contact) => {
+          console.log('Contact ', contact)
```
Comment (luthian) [inline src/components/contact/IFXContactCreateEdit.vue:55]: Remove for production.  :-)

File: src/components/contact/IFXFullContactCreateEdit.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+<template>
+  <v-container>
+    <v-form v-if='!isLoading' v-model='isValid'>
+      <v-row>
+        <v-col>
+          <v-text-field
+            v-model='item.name'
+            label='Name'
+            data-cy='name'
+            :rules='formRules.generic'
```
Comment (luthian) [inline src/components/contact/IFXFullContactCreateEdit.vue:36]: Do you want the same `formRules.name` here as you use in the previous card?

File: src/components/contact/IFXSelectableContact.vue [hunk 4]
```diff
@@ -51,68 +90,148 @@ export default {
...
           </v-autocomplete>
         </v-col>
       </v-row>
       <v-row>
         <v-col>
-          <v-text-field
-            v-model.trim="itemLocal.contact.name"
-            autocomplete="new-password"
-            :error-messages="errors['contacts.name']"
-            :rules="formRules.generic"
```
Comment (luthian) [inline src/components/contact/IFXSelectableContact.vue:68]: Do you want `formRules.name` here, too?

File: src/components/contact/IFXSelectableContact.vue [hunk 5]
```diff
@@ -51,68 +90,148 @@ export default {
...
+              ></v-text-field>
+            </v-col>
+            <v-col>
+              <v-text-field
+                v-model.trim="itemLocal.phone"
+                autocomplete="new-password"
+                :error-messages="errors['contacts.phone']"
+                label="Phone"
+                @input="checkValidForm()"
+              ></v-text-field>
```
Comment (luthian) [inline src/components/contact/IFXSelectableContact.vue:158]: Do you want `formRules.phone` here? And, perhaps, `data-cy` attributes for setting these?

File: src/components/item/IFXItemListMixin.js [hunk 6]
```diff
@@ -74,7 +74,7 @@ export default {
       return formattedItems
     },
     navigateToItemCreate() {
-      this.rtr.push({ name: `${this.itemType}Create` })
+      this.rtr.push({ name: `${this.itemType}Create`, query: { next: this.$route.path } })
```
Comment (luthian) [inline src/components/item/IFXItemListMixin.js:77]: I'm doing similar things in the branch I have for button changes.

File: src/components/item/IFXItemSelectList.vue [hunk 7]
```diff
@@ -49,6 +54,7 @@ export default {
         return this.items
       },
       set(items) {
+        console.log('changed items')
```
Comment (luthian) [inline src/components/item/IFXItemSelectList.vue:57]: Remove for production?

File: src/components/organization/IFXOrganizationCreateEdit.vue [hunk 8]
```diff
@@ -40,6 +43,8 @@ export default {
     hasItemChanged() {
       const initial = JSON.stringify(this.apiRef.decompose(this.item))
       // cachedOrganization should already be decomposed and stringified
+      console.log('initial, ', initial)
+      console.log('cached, ', this.cachedItem)
```
Comment (luthian) [inline src/components/organization/IFXOrganizationCreateEdit.vue:47]: Remove for production?

File: src/components/user/IFXUserDetail.vue [hunk 9]
```diff
@@ -33,71 +55,95 @@ export default {
...
+          <v-row align="start" dense>
+            <v-col sm="3">
+              <h3>Email(s)</h3>
+            </v-col>
+            <v-col>
+              <div class="contact-row">
+                <div class="contact-role">Primary Email</div>
+                <div class="contact-detail"><a :href="`mailto:${item.primaryEmail}`">{{item.primaryEmail}}</a></div>
+              </div>
+                <span v-html="additionalEmailList()"></span>
```
Comment (luthian) [inline src/components/user/IFXUserDetail.vue:95]: `v-html` is kind of frowned upon; I'd suggest moving the code in `additionalEmailList()` here and doing it inline

File: src/components/user/IFXUserDetail.vue [hunk 10]
```diff
@@ -33,71 +55,95 @@ export default {
...
+            </v-col>
+          </v-row>
+          <v-row dense>
+            <v-col sm="3">
+              <h3>Other Contacts</h3>
+            </v-col>
+            <v-col>
+              <span v-html="additionalContactList()"></span>
```
Comment (luthian) [inline src/components/user/IFXUserDetail.vue:103]: Same as above about moving code.

File: src/components/contact/IFXContactCreateEdit.vue (hunk 2 above)
Comment (aaronk) [reply src/components/contact/IFXContactCreateEdit.vue:55]: Done

File: src/components/contact/IFXSelectableContact.vue (hunk 4 above)
Comment (aaronk) [reply src/components/contact/IFXSelectableContact.vue:68]: Name is gone from this

File: src/components/contact/IFXSelectableContact.vue (hunk 5 above)
Comment (aaronk) [reply src/components/contact/IFXSelectableContact.vue:158]: Not really.  We talked about phone rules, but they are too complicated if you allow international.

File: src/components/contact/IFXFullContactCreateEdit.vue (hunk 3 above)
Comment (aaronk) [reply src/components/contact/IFXFullContactCreateEdit.vue:36]: Changing to formRules.generic

File: src/components/item/IFXItemListMixin.js (hunk 6 above)
Comment (aaronk) [reply src/components/item/IFXItemListMixin.js:77]: Will sort out in the merge

**Pull request merged**